packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:typed_data';
+
+import 'package:collection/collection.dart';
```

Comment:
```
   info • The imported package 'collection' isn't a dependency of the importing package • packages/flutter_tools/lib/src/commands/build_ios.dart:7:8 • depend_on_referenced_packages
```
Don't import `collection` in the tool (though it is used in the tests).  Instead of `firstOrNull` you'll need to check if it's empty explicitly:
```dart
      final Iterable<int> parsedScales = scale.split('x')
          .map((String element) => int.tryParse(element))
          .whereType<int>();
      if (parsedScales.isEmpty) {
        continue;
      }

      iconInfo[_AppIconImageFileKey(idiom, parsedSize, parsedScales.first)] = fileName;
```

Reply:
ah, it's interesting that it runs even with this dependency issue (🤯 <- me). 

I'm curious, `collection` sounds like a harmless package. Any reason why we disallow it?

Reply:
It's not disallowed, the warning is complaining `depend_on_referenced_packages` because `collection` isn't in the pubspec.  And it's not worth adding (there are implications for adding dependency version constrains in the tool) for the sake of just using the `firstOrNull` extension, which is a strange leftover from pre-null safety.